Inverted-Index/src/search/Mapper.java

```diff
@@ -0,0 +1,47 @@
+package search;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import file_handler.Doc;
+
+public class Mapper {
+
+    public Map<String, Set<Doc>> mergeMaps(Map<String, Set<Doc>> map, Map<String, Set<Doc>> doc) {
+
+        for (var pair : map.entrySet()) {
+            var key = pair.getKey();
+            if(doc.containsKey(key)) {
+                var docValues = doc.get(key);
+                var mapValues = map.get(key);
+                mapValues.addAll(docValues);
+                map.put(key, mapValues);
+            }
+        }
+
+        for (var pair : doc.entrySet()) {
+            var key = pair.getKey();
+            if(map.containsKey(key)) {
+                continue;
+            } else {
+                var set = new HashSet<Doc>();
+                set.addAll(pair.getValue());
+                map.put(key, set);
+                
+            }
+        }
+        return map;
+    }
+    public Map<String, Set<Doc>> convertDocToMap(Doc doc) {
+
+        Map<String, Set<Doc>> map = new HashMap<>();
```

Comment:
use var instead of Map<...>

Reply:
this will causes a compile error because the complier will automatically set the type of map to HashMap not Map

Reply:
"var map = new HashMap<String, Set<Doc>>();"
Are you sure that this line will cause compile error ?!
this won't cause a compile error because actually, HashMap is a child class of the Map class.

Reply:
you're right! it was my mistake